test/requests/definition_expectations_test.rb

```diff
@@ -819,6 +819,10 @@ def bar
   end
 
   def test_definition_on_self_is_disabled_for_typed_true
+    # We need this stub to make sure the test is testing the early return inside on_call_node_enter
+    # not the one inside handle_method_definition
+    RubyLsp::Listeners::Definition.any_instance.stubs(:not_in_dependencies?).returns(false)
```

Comment:
`not_in_dependencies?` doesn't seem to be getting called here (you can see by changing the `stubs` to `expects`)

Reply:
Ok, then would it be better to write `RubyLsp::Listeners::Definition.any_instance.expects(:not_in_dependencies?).never`?

Reply:
That's great, thanks!